### PR TITLE
Fix math formatting, misc edit

### LINKED ISSUE #74147: Typesetting in torch.nn.MultiheadAttention
### 📚 The doc issue

- Underscore for `num_heads` not being escaped in docstring.
https://github.com/pytorch/pytorch/blob/22b876782ffb08ef2eb8cc1799f94dc5fb60fa99/torch/nn/modules/activation.py#L1034
- Extra period after `need_weights=True`
https://github.com/pytorch/pytorch/blob/22b876782ffb08ef2eb8cc1799f94dc5fb60fa99/torch/nn/modules/activation.py#L1023

### Suggest a potential alternative/fix

- Escape underscore with `num\_heads`
- Remove extra period the code markdown

### PR DESCRIPTION
**Pull request by jaketae**
Fixes #74147.

### CODE REVIEW

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! One minor formatting thing but LGTM otherwise :)

File: torch/nn/modules/activation.py [hunk 1]
```diff
@@ -1031,7 +1031,7 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
           returns attention weights averaged across heads of shape :math:`(L, S)` when input is unbatched or
           :math:`(N, L, S)`, where :math:`N` is the batch size, :math:`L` is the target sequence length, and
           :math:`S` is the source sequence length. If ``average_weights=False``, returns attention weights per
-          head of shape :math:`(num_heads, L, S)` when input is unbatched or :math:`(N, num_heads, L, S)`.
+          head of shape :math:`(num_{heads}, L, S)` when input is unbatched or :math:`(N, num_{heads}, L, S)`.
```
[the remark this answers is not in the document]
Comment (jaketae) [reply torch/nn/modules/activation.py]: Thanks for the quick review, fixed in the most recent commit!

**Review by jbschlosser — APPROVED** [review]
LGTM!

**Pull request closed without merging**